src/vision.cs

```diff
@@ -0,0 +1,792 @@
+	[Native]
+	enum VNErrorCode : nint {
+		Ok = 0,
+		RequestCancelled,
+		InvalidFormat,
+		OperationFailed,
+		OutOfBoundsError,
+		InvalidOption,
+		IOError,
+		MissingOption,
+		NotImplemented,
+		InternalError,
+		OutOfMemory,
+		UnknownError,
+		InvalidOperation,
+		InvalidImage,
+		InvalidArgument,
+		InvalidModel,
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Native]
+	enum VNRequestTrackingLevel : nuint {
+		Accurate = 0,
+		Fast,
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Native]
+	enum VNImageCropAndScaleOption : nuint {
+		CenterCrop = 0,
+		ScaleFit = 1,
+		ScaleFill,
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	enum VNBarcodeSymbology {
+		[Field ("VNBarcodeSymbologyAztec")]
+		Aztec,
+
+		[Field ("VNBarcodeSymbologyCODE39")]
+		Code39,
+
+		[Field ("VNBarcodeSymbologyCODE39Checksum")]
+		Code39Checksum,
+
+		[Field ("VNBarcodeSymbologyCODE39FullASCII")]
+		Code39FullAscii,
+
+		[Field ("VNBarcodeSymbologyCODE39FullASCIIChecksum")]
+		Code39FullAsciiChecksum,
+
+		[Field ("VNBarcodeSymbologyCODE93")]
+		Code93,
+
+		[Field ("VNBarcodeSymbologyCODE93i")]
+		Code93i,
+
+		[Field ("VNBarcodeSymbologyCODE128")]
+		Code128,
+
+		[Field ("VNBarcodeSymbologyDataMatrix")]
+		DataMatrix,
+
+		[Field ("VNBarcodeSymbologyEAN8")]
+		Ean8,
+
+		[Field ("VNBarcodeSymbologyEAN13")]
+		Ean13,
+
+		[Field ("VNBarcodeSymbologyI2OF5")]
+		I2OF5,
+
+		[Field ("VNBarcodeSymbologyI2OF5Checksum")]
+		I2OF5Checksum,
+
+		[Field ("VNBarcodeSymbologyITF14")]
+		Itf14,
+
+		[Field ("VNBarcodeSymbologyPDF417")]
+		Pdf417,
+
+		[Field ("VNBarcodeSymbologyQR")]
+		QR,
+
+		[Field ("VNBarcodeSymbologyUPCE")]
+		Upce,
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface VNCoreMLModel {
+
+		[Static]
+		[Export ("modelForMLModel:error:")]
+		[return: NullAllowed]
+		VNCoreMLModel FromMLModel (MLModel model, out NSError error);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	delegate void VNRequestCompletionHandler (VNRequest request, NSError error);
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNImageBasedRequest))]
+	[DisableDefaultCtor]
+	interface VNCoreMLRequest {
+
+		[Export ("model")]
+		VNCoreMLModel Model { get; }
+
+		[Export ("imageCropAndScaleOption", ArgumentSemantic.Assign)]
+		VNImageCropAndScaleOption ImageCropAndScaleOption { get; set; }
+
+		[Export ("initWithModel:")]
+		IntPtr Constructor (VNCoreMLModel model);
+
+		[Export ("initWithModel:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (VNCoreMLModel model, [NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithCompletionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([NullAllowed] VNRequestCompletionHandler completionHandler);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Abstract]
```

Comment:
That will still generate a
`protected VNDetectBarcodesRequest ();`
which should not be used in subclasses as there is a designated initializer defined for the type (unless the type has `init` defined as one)